Sources/Memberlist.php

```diff
@@ -304,7 +304,11 @@ function MLAll()
 	if ($use_cache && $_REQUEST['sort'] === 'real_name' && !isset($_REQUEST['desc']))
 	{
 		$first_offset = $_REQUEST['start'] - ($_REQUEST['start'] % $cache_step_size);
+		if ($first_offset < 0)
+			$first_offset = 0;
 		$second_offset = ceil(($_REQUEST['start'] + $modSettings['defaultMaxMembers']) / $cache_step_size) * $cache_step_size;
+		if ($second_offset >= $memberlist_cache['num_members'])
+			$second_offset = $memberlist_cache['num_members'] - 1;
```

Comment:
```suggestion
		$second_offset = min($memberlist_cache['num_members'] - 1, ceil(($_REQUEST['start'] + $modSettings['defaultMaxMembers']) / $cache_step_size) * $cache_step_size);
```

Reply:
Again, I disagree...  It is not at all clear to see what you're aiming at here...  The suggested is very simple...   

Syntax-wise, 6 of one, half dozen of the other.  Readability-wise, I think we should not make the change.